crypto/signatures_test.go

```diff
@@ -2,24 +2,138 @@ package crypto
 
 import (
 	"crypto/rand"
+	"errors"
 	"testing"
 
 	"github.com/NebulousLabs/Sia/encoding"
 )
 
-// Creates and encodes a public key, and verifies that it decodes correctly,
-// does the same with a signature.
-func TestSignatureEncoding(t *testing.T) {
+// mockEntropySource is a mock implementation of entropySource that allows the
+// client to specify the returned entropy and error.
+type mockEntropySource struct {
+	called  bool
+	entropy [EntropySize]byte
+	err     error
+}
+
+func (es *mockEntropySource) getEntropy() (entropy [EntropySize]byte, err error) {
+	es.called = true
+	return es.entropy, es.err
+}
+
+// mockKeyDeriver is a mock implementation of keyDeriver that saves its provided
+// entropy and allows the client to specify the returned SecretKey and
+// PublicKey.
+type mockKeyDeriver struct {
+	called  bool
+	entropy [EntropySize]byte
+	sk      SecretKey
+	pk      PublicKey
+}
+
+func (kd *mockKeyDeriver) deriveKeyPair(entropy [EntropySize]byte) (sk SecretKey, pk PublicKey) {
+	kd.called = true
+	kd.entropy = entropy
+	return kd.sk, kd.pk
+}
+
+// Test that the Generate method is properly calling its dependencies and
+// returning the expected key pair.
+func TestGenerateRandomKeyPair(t *testing.T) {
+	var mockEntropy [EntropySize]byte
+	mockEntropy[0] = 0x0a
+	mockEntropy[EntropySize-1] = 0x0b
+	es := mockEntropySource{entropy: mockEntropy}
+
+	sk := SecretKey([SecretKeySize]byte{})
+	pk := PublicKey([PublicKeySize]byte{})
```

Comment:
This no longer applies because the code has changed, but I think that's probably true. I'm still getting used to Go's typing / auto initialization / conversions.